### PR TITLE
Fix Issue #4755 - Swift 5 Migration

### PR DESCRIPTION
**Pull request by q2r5**
Updates dependencies and code to Swift 5.

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-github "adjust/ios_sdk"                     "v4.17.1"
+github "kif-framework/KIF"                  "v3.7.7"
+github "adjust/ios_sdk"                     "v4.17.2"
 github "AgileBits/onepassword-app-extension"    "add-framework-support"
-github "mozilla-mobile/SwiftKeychainWrapper"       "master" # switch to mozilla-mobile once PR#1 lands there
-github "DaveWoodCom/XCGLogger"              ~> 6.1
-github "cezheng/Fuzi"                       "master"
-github "SwiftyJSON/SwiftyJSON"              ~> 4.3
-github "st3fan/EarlGrey"                    "master"
+github "mozilla-mobile/SwiftKeychainWrapper"       "develop"
```
Comment (farhanpatel) [inline Cartfile]: Can we update to the official jrendel/SwiftKeychainWrapper?

File: Cartfile.resolved [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+github "cezheng/Fuzi" "3.0.0"
+github "getsentry/sentry-cocoa" "4.3.4"
+github "google/EarlGrey" "1.15.1"
+github "kif-framework/KIF" "v3.7.7"
+github "mozilla-mobile/MappaMundi" "02b6f0b404d0a7178c47c073550936016f42981e"
 github "mozilla-mobile/SwiftKeychainWrapper" "6404351541ce1c0b07652f81a0a20097b7f86297"
 github "mozilla-mobile/telemetry-ios" "1.1.1"
-github "rs/SDWebImage" "4.4.6"
-github "st3fan/EarlGrey" "ce90f3ffabfb9cfcda73bcbcdd8844c47fede04e"
+github "mozilla/application-services" "v0.27.2"
```
Comment (farhanpatel) [inline Cartfile.resolved]: application services need to be 0.28.0

File: Client/Frontend/Browser/MetadataParserHelper.swift [hunk 3]
```diff
@@ -84,6 +82,6 @@ class MediaImageLoader: TabEventHandler {
     }
 
     fileprivate func cache(image: UIImage, forURL url: URL) {
-        SDWebImageManager.shared().saveImage(toCache: image, for: url)
+        SDImageCache.shared.storeImage(toMemory: image, forKey: url.absoluteString)
```
Comment (farhanpatel) [inline Client/Frontend/Browser/MetadataParserHelper.swift]: I think `SDWebImageManager.shared().saveImage(toCache` was saving this image to the disk if I'm not mistaken

File: Client/Frontend/Home/FirefoxHomeTopSitesCell.swift [hunk 4]
```diff
@@ -79,7 +79,7 @@ class TopSiteItemCell: UICollectionViewCell, Themeable {
         accessibilityIdentifier = "TopSite"
         contentView.addSubview(titleLabel)
         contentView.addSubview(faviconBG)
-        contentView.addSubview(imageView)
+        faviconBG.addSubview(imageView)
```
Comment (farhanpatel) [inline Client/Frontend/Home/FirefoxHomeTopSitesCell.swift]: curious why this layout needed to change?

Comment (q2r5) [reply Client/Frontend/Home/FirefoxHomeTopSitesCell.swift]: https://github.com/SnapKit/SnapKit/pull/569 changed how insets and centering is applied. This change just re-centers the favicon on the background.

**Comment by q2r5** [comment]
Fixed and ready for another review.

**Pull request merged**